File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-4.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        n = len(nums)
        lo, hi = 0, n-1

        if nums[0] <= nums[n-1]:
            return nums[0]

        while lo < hi:
            mid = lo + (hi-lo)//2

            if nums[n-1] >= nums[mid]:
                hi = mid
            else:
                lo = mid + 1
        
        return nums[hi]
            

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int n = nums.size();
        unordered_map<int, int> m;
        int res = 0;

        for(int i:nums)
            m[i]++;

        for(int i=0; i<n; i++){
            // start of seq detected
            if(m.find(nums[i]-1)==m.end()){
                int len = 0;
                while(m.find(nums[i]+len)!=m.end()){
                    len++;
                }
                res = max(res, len);
            }
        }
        
        return res;
    }
};
